bin/git-theta

```diff
@@ -39,49 +44,64 @@ def parse_args():
+    )
+    add_parser.add_argument("--update-type", choices=["dense", "sparse"], help="Type of update being applied")
+    add_parser.set_defaults(func=add)
+
+    args = parser.parse_known_args()
     return args
 
 
-def paths_and_parameters(model_dict, root):
-    """Convert the keys of the model dict to parameters based starting at root."""
-    return {os.path.join(root, *k): v for k, v in utils.flatten(model_dict).items()}
+def post_commit(args):
+    repo = git_utils.get_git_repo()
+    theta_commits = theta.ThetaCommits(repo)
+
+    gitattributes_file = git_utils.get_gitattributes_file(repo)
+    patterns = git_utils.get_gitattributes_tracked_patterns(gitattributes_file)
+
+    oids = set()
+    commit = repo.commit('HEAD')
+    for obj in commit.tree.traverse():
+        if any([fnmatch.fnmatchcase(obj.path, pattern) for pattern in patterns]):
+            curr_metadata = metadata.Metadata.from_file(obj.data_stream).flatten()
+
+            prev_obj = git_utils.get_file_version(repo, obj.path, 'HEAD~1')
+            if prev_obj:
+                prev_metadata = metadata.Metadata.from_file(prev_obj.data_stream).flatten()
+            else:
+                prev_metadata = metadata.Metadata()
+
+            added, removed, modified = curr_metadata.diff(prev_metadata)
+            oids.update([param.lfs_metadata.oid for param in added + modified])
 
+    theta_commits.write_commit_oids(commit.hexsha, oids)
 
-def add(args):
+
+def pre_push(args):
     """
-    Splits model parameters file into parameter groups on filesystem and stages files
+    Pre-push git hook for sending objects to the LFS server
     """
-    logging.debug(f"Running git theta add on {args.file}")
     repo = git_utils.get_git_repo()
-    model_path = git_utils.get_relative_path_from_root(repo, args.file)
-    theta_model_dir = git_utils.get_git_theta_model_dir(repo, model_path, create=True)
-
-    checkpoint_handler = checkpoints.get_checkpoint_handler(args.type)
-    logging.debug(f"git theta add using checkpoint type={checkpoint_handler.name}")
-    param_dict = checkpoint_handler.from_file(args.file)
-    
-    for param_dir, param in paths_and_parameters(param_dict, theta_model_dir).items():
-        param_file = os.path.join(param_dir, "params")
-        os.makedirs(param_file, exist_ok=True)
-        file_io.write_tracked_file(param_file, param)
-        git_utils.add_file(param_file, repo)
+    theta_commits = theta.ThetaCommits(repo)
 
-    for param_file in utils.removed_params(param_dict, utils.walk_parameter_dir(theta_model_dir)):
-        git_utils.remove_file(param_file, repo)
-
-    git_utils.add_file(model_path, repo)
+    # Read lines of the form <local ref> <local sha1> <remote ref> <remote sha1> LF
+    lines = sys.stdin.readlines()
+    print(lines)
+    lines_parsed = [re.match('^(?P<local_ref>[^\s]+)\s+(?P<local_sha1>[a-f0-9]{40})\s+(?P<remote_ref>[^\s]+)\s+(?P<remote_sha1>[a-f0-9]{40})\s+$', l) for l in lines]
```

Comment:
We should abstract this parsing into a function that we can test.

Reply:
Done.